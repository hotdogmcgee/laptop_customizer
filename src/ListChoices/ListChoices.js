import React from 'react'
import ListItem from '../ListItem/ListItem.js'

class ListChoices extends React.Component {
    constructor(props) {
        super(props)
    }

    render() {
        const { features, currentItem } = this.props

        const displayFeatures = Object.keys(features)
          .map((featuresItem, index) => {
            
              return <ListItem 
                featureName={featuresItem} 
                featuresItem={features[featuresItem]}
                updateFeature={this.props.updateFeature}
                key={index}
                currentItem={currentItem}
                />
            }); 
    

    return (
        <div className="List-choices-container">
                {displayFeatures}
        </div>
        )
    }
}

ListChoices.defaultProps = {
    features: {},
    currentItem: []
}

export default ListChoices
