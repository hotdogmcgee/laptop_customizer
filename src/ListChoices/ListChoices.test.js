import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListChoices from './ListChoices.js'


describe('ListChoices component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListChoices />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const FEATURES = {
        Processor: [
          {
            name: '17th Generation Intel Core HB (7 Core with donut spare)',
            cost: 700
          },
          {
            name: 'Professor X AMD Fire Breather with sidewinder technology',
            cost: 1200
          }
        ],
        "Operating System": [
          {
            name: 'Ubuntu Linux 16.04',
            cost: 200
          },
          {
            name: 'Bodhi Linux',
            cost: 300
          }
        ],
        "Video Card": [
          {
            name: 'Toyota Corolla 1.5v',
            cost: 1150.98
          },
          {
            name: 'Mind mild breeze 2000',
            cost: 1345
          }
        ],
        Display: [
          {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
          },
          {
            name: '15.3" HGTV (3840 x 2160) Home makeover edition',
            cost: 1400
          },
        ]
      };
    const tree = renderer
      .create(<ListChoices features={FEATURES} updateFeature={() => {return 1}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});