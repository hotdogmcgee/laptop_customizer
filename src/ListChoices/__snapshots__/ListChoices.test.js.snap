// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListChoices component renders the UI as expected 1`] = `
<div
  className="List-choices-container"
>
  <div
    className="feature"
  >
    <div
      className="feature__name"
    />
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            17th Generation Intel Core HB (7 Core with donut spare)
          </div>
          (
          $700.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            Professor X AMD Fire Breather with sidewinder technology
          </div>
          (
          $1,200.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
  >
    <div
      className="feature__name"
    />
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            Ubuntu Linux 16.04
          </div>
          (
          $200.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            Bodhi Linux
          </div>
          (
          $300.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
  >
    <div
      className="feature__name"
    />
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            Toyota Corolla 1.5v
          </div>
          (
          $1,150.98
          )
        </div>
      </li>
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            Mind mild breeze 2000
          </div>
          (
          $1,345.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
  >
    <div
      className="feature__name"
    />
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
          </div>
          (
          $1,500.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        onClick={[Function]}
      >
        <div
          className="feature__option "
        >
          <div>
            15.3" HGTV (3840 x 2160) Home makeover edition
          </div>
          (
          $1,400.00
          )
        </div>
      </li>
    </ul>
  </div>
</div>
`;
